python/pyspark/sql/catalog.py

```diff
@@ -27,7 +27,7 @@
 
 if TYPE_CHECKING:
     from pyspark.sql._typing import UserDefinedFunctionLike
-    from pyspark.sql.types import DataType
+    from pyspark.sql.connect._typing import DataTypeOrString
```

Comment:
I guess we should not import `DataTypeOrString` from connect side?
```suggestion
    from pyspark.sql._typing import DataTypeOrString
```

Reply:
Thanks for the correction :-)